src/tools/manifest2proto.ts

```diff
@@ -150,7 +150,26 @@ export async function typeToProtoPayload(type: Type) {
         referredType: await typeToProtoPayload((type as ReferenceType<Type>).referredType)
       }
     };
-    default: throw Error(`Type ${type.tag} is not supported.`);
+    case 'Tuple': return {
+      tuple: {
+        elements: await Promise.all((type as TupleType).innerTypes.map(typeToProtoPayload))
+      }
+    };
+    case 'Singleton': return {
+      singleton: {
+        singletonType: await typeToProtoPayload((type as SingletonType<Type>).getContainedType())
+      }
+    };
+    case 'Count': return {
+      count: {}
+    };
+    // TODO(b/154733929)
+    // case 'TypeVariable': return {
```

Comment:
As I understand it, supporting TypeVariables now means that the incoming recipe would not be fully resolved. 

This type needs design attention.

Reply:
For Plans it will be important that we serialize the resolution of the type variable - if available.